tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1146,16 +1147,18 @@ unsigned CGMSHLSLRuntime::ConstructStructAnnotation(DxilStructAnnotation *annota
       if (m_PreciseOutputSet.count(StringRef(fieldSemName).lower()))
         fieldAnnotation.SetPrecise();
     }
-
     // Update offset.
     CBufferSize = std::max(CBufferSize, CBufferOffset + size);
     CBufferOffset = CBufferSize;
 
     ++Field;
   }
 
+  if (RD->isUnion() && CBufferSize == 0)
```

Comment:
How do we handle empty structs here? Unions and structs should be interchangeable in this case.